server/src/main/java/org/opensearch/gateway/ShardsBatchGatewayAllocator.java

```diff
@@ -198,6 +198,17 @@ protected ShardsBatchGatewayAllocator(long batchSize) {
         this.replicaShardsBatchGatewayAllocatorTimeout = null;
     }
 
+    protected ShardsBatchGatewayAllocator(RerouteService rerouteService) {
+        this.rerouteService = rerouteService;
+        this.batchStartedAction = null;
+        this.primaryShardBatchAllocator = null;
+        this.batchStoreAction = null;
+        this.replicaShardBatchAllocator = null;
+        this.maxBatchSize = DEFAULT_SHARD_BATCH_SIZE;
+        this.primaryShardsBatchGatewayAllocatorTimeout = null;
+        this.replicaShardsBatchGatewayAllocatorTimeout = null;
+    }
+
```

Comment:
Can we create a common ctor 
```
protected ShardsBatchGatewayAllocator(RerouteService rerouteService, long batchSize) {
        this.rerouteService = rerouteService;
        this.batchStartedAction = null;
        this.primaryShardBatchAllocator = null;
        this.batchStoreAction = null;
        this.replicaShardBatchAllocator = null;
        this.maxBatchSize = batchsize;
        this.primaryShardsBatchGatewayAllocatorTimeout = null;
        this.replicaShardsBatchGatewayAllocatorTimeout = null;
    }
```

and the current ctor just do this(blah, blah...) and invoke the common ctor

Reply:
Updated